File: examples_android/app/src/main/java/com/example/myapp/domain/ProgramConfig.kt
package com.example.myapp.domain

import com.example.myapp.ConfigKey
import com.example.myapp.utils.Utils
import java.io.Serializable

class ProgramConfig(val serverId: String, val programKey: String) : Serializable {
    var customData: String
        get() {
            return get(ConfigKey.CUSTOM_DATA, "")
        }
        set(value) {
            set(ConfigKey.CUSTOM_DATA, value)
        }

    private inline fun <reified T> get(key: ConfigKey, def: T): T {
        return Utils.getSetting(createKey(key), def)
    }

    private inline fun <reified T> set(key: ConfigKey, value: T) {
        Utils.setSetting(createKey(key), value)
    }

    private fun createKey(key: ConfigKey): String {
        return "${serverId}_${programKey}_${key.name}"
    }
}

File: examples_ios/MyApp/Sources/Domain/ProgramConfig.swift
import Foundation

enum ProgramConfigKey: String {
  case customData
}

class ProgramConfig {
  let serverId: String
  let programKey: String

  init(serverId: String, programKey: String) {
    self.serverId = serverId
    self.programKey = programKey
  }

  var customData: String {
    get {
      return get(.customData, "")
    }
    set {
      set(.customData, newValue)
    }
  }

  private func get<T>(_ keyEnum: ProgramConfigKey, _ def: T) -> T {
    return Utils.getSetting(key(keyEnum), def)
  }

  private func set<T>(_ keyEnum: ProgramConfigKey, _ value: T) {
    Utils.setSeting(key(keyEnum), value)
  }

  private func key(_ key: ProgramConfigKey) -> String {
    return "\(serverId)_\(programKey)_\(key.rawValue)"
  }
}
